### PR TITLE
HLD : Query parameter for get request

### PR DESCRIPTION
**Pull request by s-mari**

### CODE REVIEW

File: SONiC_QueryParameterForGet_HLD.md [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
...a+json"<br />     **Response:**  {<br />     "openconfig-network-instance-   deviation:igmpsnooping":    {<br />               "interfaces": {<br />                   "interface": [{<br />                       "config": {<br />                           "enabled": true,<br />                            "last-memberquery-interval": 2000,                            "name": "Vlan1"<br />                      },<br />                      "name": "Vlan1",<br />                      "state": {<br />                          "enabled": true,<br />                         "last-memberquery-interval": 2000,<br />                         "name": "Vlan1"<br />                      }<br />                  }, {<br />                       "config": {<br />                          "enabled": false,<br />                           "last-memberquery-interval": 4000,<br />         "name": "Vlan2",<br />         "version": 3<br />       },<br />       "name": "Vlan2",<br />       "state": {<br />         "enabled": false,<br />         "last-memberquery-interval":  4000,<br />         "name": "Vlan2",<br />         "version": 3<br />       }<br />    }]<br />              }<br />    }<br />  } |
```
Comment (alacaya) [inline SONiC_QueryParameterForGet_HLD.md]: If a field belongs to a list, should it retrieve the field from all entries of the lists?

**Comment by a-barboza** [comment]
@s-mari @jeff-yin A general comment: There appear to be 2 files with approximately the same content. One is in the root directory of the repository, whereas the other is in the subdirectory at "doc/mgmt". Could we have only 1 document, otherwise there might be confusion with regards to the authoritative version?

File: SONiC_QueryParameterForGet_HLD.md (hunk 1 above)
Comment (kwangsuk) [reply SONiC_QueryParameterForGet_HLD.md]: Yes, the server returns the selected data nodes from all entries.
e.g. using sub-selector with target resource list
GET /restconf/data/openconfig-interfaces:interfaces/interface?fields=state(name;admin-status;oper-status;counters)

**Comment by kwangsuk** [comment]
> @s-mari @jeff-yin A general comment: There appear to be 2 files with approximately the same content. One is in the root directory of the repository, whereas the other is in the subdirectory at "doc/mgmt". Could we have only 1 document, otherwise there might be confusion with regards to the authoritative version?

Thanks @a-barboza. As commented, the file in the top folder is now removed. The doc/mgmt folder has an active HLD for QP.

**Review by jeffyin-dell — APPROVED** [review]
Approved. Will merge this into the `master` branch due to lack of recent/open comments. Any follow-up can be addressed in subsequent doc PRs.

**Pull request merged**